weave

```diff
@@ -575,9 +576,18 @@ wait_for_status_ip() {
     done
 }
 
-wait_for_status() {
-    container_ip $1 "$1 container has died." "$1 container has died." || return 1
-    wait_for_status_ip $1 $CONTAINER_IP $2
+# Wait until container $1 outputs a line containing $2 in its log
+wait_for_log() {
+    fifo=/tmp/tmpfifo.$$
+    mkfifo $fifo || exit 1
+    docker logs -f $1 2>/dev/null >$fifo &
```

Comment:
Ah, it gets killed when the container exits, so there's probably no point removing the fifo either.